### PR TITLE
Expose the logic for extracting error messages

### PR DESCRIPTION
**Pull request by chadly**
The library now exports `extractErrorMessage`, the previously internal functionality to pull out error messages from server data. Need this to make research display better error messages from legacy jquery ajax functions which don't use real promises.

### CODE REVIEW

**Comment by chadly** [comment]
Tested locally within the research app

File: test/extracting-error-messages.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+	it("should just return string as the error message", function() {
+		const msg = extractErrorMessage("oh, hai");
+		expect(msg).to.equal("oh, hai");
+	});
+
+	it("should not extract the error message from empty object", function() {
+		const msg = extractErrorMessage({});
+		expect(msg).to.not.be.ok;
+	});
```
Comment (broline) [inline test/extracting-error-messages.js]: What about a test for 
```js
{
    exceptionMessage: "poopity scoop"
}
```
or

```js
{
    ExceptionMessage: "poopity scoop"
}
```

or

```js
{
    Message: "poopity scoop"
}
```

Comment (chadly) [reply test/extracting-error-messages.js]: A lot of those test cases already exist in [`parsing-response.js`](https://github.com/civicsource/fetch-helpers/blob/master/test/parsing-response.js). See this one:

https://github.com/civicsource/fetch-helpers/blob/e395d385941105801735cc80f5502c5824e45a50/test/parsing-response.js#L194-L212

File: src/extract-error-message.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		"exceptionMessage",
+		get(ex, "message", get(ex, "ExceptionMessage", get(ex, "Message")))
+	);
+
+export default data => {
+	if (isString(data)) return data;
+	if (!data) return null;
+
+	// if it is a .net exception, try to find the inner-most exception
+	let ex = data;
```
Comment (mdarens) [inline src/extract-error-message.js]: nice

**Pull request merged**